packages/rest/src/__tests__/integration/export-openapi-spec.integration.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright IBM Corp. 2019,2020. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {ApplicationConfig} from '@loopback/core';
+import {expect, givenHttpServerConfig, TestSandbox} from '@loopback/testlab';
+import fs from 'fs';
+import {safeLoad} from 'js-yaml';
+import path from 'path';
+import {format} from 'util';
+import {get, RestApplication} from '../..';
+
+const sandbox = new TestSandbox(path.resolve(__dirname, '../../../.sandbox'));
+
+describe('exportOpenApiSpec', () => {
+  let app: MyApp;
+  let lastLog = '';
+  let bootCalled = false;
+
+  const log = (formatter: unknown, ...args: unknown[]) => {
+    lastLog = format(formatter, ...args);
+  };
+
+  const expectedSpec = {
+    openapi: '3.0.0',
+    info: {
+      title: 'LoopBack Application',
+      version: '1.0.0',
+    },
+    paths: {
+      '/hello': {
+        get: {
+          'x-controller-name': 'MyController',
+          'x-operation-name': 'hello',
+          tags: ['MyController'],
+          responses: {
+            '200': {
+              description: 'Return value of MyController.hello',
+            },
+          },
+          operationId: 'MyController.hello',
+        },
+      },
+    },
+    servers: [
+      {
+        url: '/',
+      },
+    ],
+  };
+
+  beforeEach(async () => {
+    lastLog = '';
+    bootCalled = false;
+    await givenApp();
+  });
+
+  it('saves the spec to a json file for RestApplication', async () => {
+    const file = path.join(sandbox.path, 'openapi.json');
+    await app.exportOpenApiSpec(file, log);
+    expectJsonSpec(file);
+  });
+
+  it('saves the spec to a json file for RestServer', async () => {
+    const file = path.join(sandbox.path, 'openapi.json');
+    await app.restServer.exportOpenApiSpec(file, log);
+    expectJsonSpec(file);
+  });
+
+  it('writes the spec as a json document to console', async () => {
+    await app.exportOpenApiSpec('-', log);
+    expect(JSON.parse(lastLog)).to.eql(expectedSpec);
+  });
+
+  it('saves the spec to a yaml file', async () => {
+    const file = path.join(sandbox.path, 'openapi.yaml');
+    await app.restServer.exportOpenApiSpec(file, log);
+    expect(lastLog.match(/The OpenAPI spec is saved to .+ openapi\.yaml$/));
+    const content = fs.readFileSync(file, 'utf-8');
+    expect(safeLoad(content)).to.eql(expectedSpec);
+    expect(bootCalled).to.be.true();
```

Comment:
I find `bootCalled` checks redundant. The test is relying on a beforeEach hook to setup the application and boot it up, and the test does not call any boot-related APIs. Why would ever `bootCalled` be false?